lib/ModuleManager.cpp

```diff
@@ -839,11 +840,82 @@ void QoreModuleManager::loadModuleIntern(ExceptionSink& xsink, const char* name,
         ++w;
     }
 
+    // Qore 0.9.0
+    if (tryToLoadAsModuleDir(xsink, name, pgm, reexport, op, version, src, mpgm, load_opt))
+        return;
+
     QoreStringNode* desc = new QoreStringNodeMaker("feature '%s' is not builtin and no module with this name could be found in the module path: ", name);
     moduleDirList.appendPath(*desc);
     xsink.raiseExceptionArg("LOAD-MODULE-ERROR", new QoreStringNode(name), desc);
 }
 
+bool QoreModuleManager::tryToLoadAsModuleDir(
+    ExceptionSink& xsink,
+    const char* name,
+    QoreProgram* pgm,
+    bool reexport,
+    mod_op_e op,
+    version_list_t* version,
+    const char* src,
+    QoreProgram* mpgm,
+    unsigned load_opt) {
+
+    strdeque_t::const_iterator path = moduleDirList.begin();
+    while (path != moduleDirList.end()) {
+        QoreString modulePath(*path);
+        modulePath += QORE_DIR_SEP_STR;
+        modulePath += name;
+
+        QoreDir modulDir(&xsink, QCS_DEFAULT, modulePath.c_str());
+        if (xsink)
+            return false;
+
+        if (!modulDir.checkPath()) {
+            modulePath += QORE_DIR_SEP_STR;
+            modulePath += name;
+            modulePath += ".qm";
+
+            loadModuleIntern(xsink, modulePath.c_str(), pgm, reexport, op, version, src, mpgm, load_opt);
+            if (!xsink)
+                return true;
+        }
+        ++path;
+    }
+    return false;
+}
+
+bool exists(const QoreString& filePath) {
```

Comment:
Should it be static?

Reply:
yes, please make it static for now and change the name to `exists_file()` or `exists_path()` or something like that

Reply:
Is it ok to move it to QoreDir?

Reply:
Is it ok to move it to QoreDir?

Reply:
yes, sure, good idea